test/test_jit_fuser_te.py

```diff
@@ -148,6 +148,21 @@ def test_abs_cpu(self):
     def test_abs_cuda(self):
         self._test_fused_abs(device="cuda")
 
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    def test_unsqueeze_size_calculation(self):
+
+        def foo(b, d):
+            x = d.unsqueeze(1)
+            y = x * 42.
+            z = b + y
+            r = z / 42.
+            return r
+
+        inputs = (torch.rand(20, 28, device='cuda', requires_grad=True), torch.rand(20, device='cuda'))
```

Comment:
Can you rewrite this without requiring autograd? It's really difficult to see what exactly your example fixed. You can rewrite whichever graph was generated in the backward and just use that for testing.